src/pex-builder/build-builder.sh

```diff
@@ -26,7 +26,7 @@ python3.8 -m venv $VENV_ROOT
 source $VENV_ROOT/bin/activate
 
 export PIPENV_IGNORE_VIRTUALENVS=1
-$VENV_ROOT/bin/pip install pipenv "pex<=2.1.111"
+$VENV_ROOT/bin/python -m pip install pipenv pex
```

Comment:
Don't need to pin pex anymore since their file locking issue is fixed.

Reply:
Also switched to `python -m pip` because for some unknown reason my local venvs are not getting a pip binary by default.